### PR TITLE
Handle `batch/v1` CronJobs and Update `envtest` to use `1.25`

### PR DESCRIPTION
**Pull request by shafeeqes**
**How to categorize this PR?**

/area testing
/kind task

**What this PR does / why we need it**:
The garbage collector in the GRM was trying to list batch/v1beta1 CronJobs which are not present in Kubernetes v1.25.  This PR adds handling for this.

Also this PR updates envtest to use version 1.25.0.
The workaround for `arm64` architecture introduced in https://github.com/gardener/gardener/pull/6027 is no longer needed, because now we have releases for arm64 as well, see 
https://github.com/kubernetes-sigs/kubebuilder/issues/2664 

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:
/cc @shreyas-s-rao 

**Release note**:

```other operator
NONE
```

### CODE REVIEW

File: pkg/resourcemanager/controller/garbagecollector/reconciler.go [hunk 1]
```diff
@@ -83,15 +86,33 @@ func (r *Reconciler) Reconcile(reconcileCtx context.Context, _ reconcile.Request
...
+			batchv1.SchemeGroupVersion.WithKind("JobList"),
+			corev1.SchemeGroupVersion.WithKind("PodList"),
+		}
+	)
+
+	k8sGreater121, err := versionutils.CheckVersionMeetsConstraint(r.TargetKubernetesVersion.String(), ">= 1.21")
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+	k8sLesser125, err := versionutils.CheckVersionMeetsConstraint(r.TargetKubernetesVersion.String(), "< 1.25")
```
Comment (rfranzke) [inline pkg/resourcemanager/controller/garbagecollector/reconciler.go]:
```suggestion
	k8sLess125, err := versionutils.CheckVersionMeetsConstraint(r.TargetKubernetesVersion.String(), "< 1.25")
```

File: pkg/resourcemanager/controller/add.go [hunk 2]
```diff
@@ -43,6 +44,16 @@ func AddToManager(mgr manager.Manager, sourceCluster, targetCluster cluster.Clus
...
 
+	targetServerVersion, err := targetClientSet.Discovery().ServerVersion()
+	if err != nil {
+		return err
+	}
+
+	targetKubernetesVersion, err := semver.NewVersion(targetServerVersion.GitVersion)
+	if err != nil {
+		return err
+	}
```
Comment (rfranzke) [inline pkg/resourcemanager/controller/add.go]: What is the benefit of parsing the semver here if you use `versionutils.CheckVersionMeetsConstraint` with the string representation later on?

**Review by acumino — COMMENTED** [review]
/lgtm

**Review by rfranzke — COMMENTED** [review]
/approve

File: pkg/operation/botanist/cleanup.go [hunk 3]
```diff
@@ -249,13 +249,20 @@ func (b *Botanist) CleanKubernetesResources(ctx context.Context) error {
...
+
 	if version.ConstraintK8sLess119.Check(b.Shoot.KubernetesVersion) {
 		ingressList = &extensionsv1beta1.IngressList{}
 	}
+	if version.ConstraintK8sGreaterEqual121.Check(b.Shoot.KubernetesVersion) {
+		cronJobList = &batchv1.CronJobList{}
+	}
 
 	return flow.Parallel(
-		cleanResourceFn(ops, c, &batchv1beta1.CronJobList{}, CronJobCleanOption, cleanOptions),
```
Comment (acumino) [inline pkg/operation/botanist/cleanup.go:258]: for Kubernetes version <1.25 `batchv1beta1.CronJobList{}` can be there ?

Comment (shafeeqes) [reply pkg/operation/botanist/cleanup.go:258]: From https://kubernetes.io/docs/reference/using-api/deprecation-guide/#cronjob-v125,
> All existing persisted objects are accessible via the new API

**Review by rfranzke — COMMENTED** [review]
/lgtm

**Pull request merged**